nixos/modules/services/databases/neo4j.nix

```diff
@@ -36,81 +89,412 @@ let
     dbms.jvm.additional=-XX:+TrustFinalNonStaticFields
     dbms.jvm.additional=-XX:+DisableExplicitGC
     dbms.jvm.additional=-Djdk.tls.ephemeralDHKeySize=2048
-
+    dbms.jvm.additional=-Djdk.tls.rejectClientInitiatedRenegotiation=true
     dbms.jvm.additional=-Dunsupported.dbms.udc.source=tarball
+
+    # Usage Data Collector
+    dbms.udc.enabled=${boolToString cfg.udc.enable}
+
+    # Extra Configuration
+    ${cfg.extraServerConfig}
   '';
 
 in {
 
   ###### interface
 
   options.services.neo4j = {
-    enable = mkOption {
-      description = "Whether to enable neo4j.";
+
+    allowUpgrade = mkOption {
+      type = types.bool;
       default = false;
+      description = ''
+        Allow upgrade of Neo4j database files from an older version.
+      '';
+    };
+
+    constrainLoadCsv = mkOption {
       type = types.bool;
+      default = true;
+      description = ''
+        <para>Sets the root directory for file URLs used with the Cypher
+        <literal>LOAD CSV</literal> clause to be that defined by
+        <option>directories.imports</option>. It restricts
+        access to only those files within that directory and its
+        subdirectories.</para>
+
+        <para>Setting this option to <literal>false</literal> introduces
+        possible security problems.</para>
+      '';
+    };
+
+    defaultListenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        <para>Default network interface to listen for incoming connections. To
+        listen for connections on all interfaces, use "0.0.0.0".</para>
+
+        <para>Specifies the default IP address and address part of connector
+        specific <option>listenAddress</option> options. To bind specific
+        connectors to a specific network interfaces, specify the entire
+        <option>listenAddress</option> option for that connector.</para>
+      '';
+    };
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable Neo4j Community Edition.
+      '';
+    };
```

Comment:
Nixpick: `enable` should be the topmost option. (sidenote: The manual also uses this special ordering rule when building the manual, so you'll always find `enable` options at the top)

Also, you could use `mkEnableOption "Neo4j Community Edition"`